integrationtests/src/test/java/org/neo4j/procedure/ProcedureIT.java

```diff
@@ -523,6 +538,77 @@ public void shouldGracefullyFailWhenSpawningThreadsCreatingTransactionInProcedur
         }
     }
 
+    @Test
+    public void shouldBeAbleToUseCallYieldWithPeriodicCommit() throws IOException
+    {
+        // GIVEN
+        String[] lines = IntStream.rangeClosed( 1, 100 )
+                .boxed()
+                .map( i -> Integer.toString( i ) )
+                .toArray( String[]::new );
+        String url = createCsvFile( lines);
+
+        //WHEN
+        Result result = db.execute( "USING PERIODIC COMMIT 1 " +
+                                    "LOAD CSV FROM '" + url + "' AS line " +
+                                    "CALL org.neo4j.procedure.createNode(line[0]) YIELD node as n " +
+                                    "RETURN n.prop" );
+        // THEN
+        for (int i = 1; i <= 100; i++)
+        {
+            assertThat( result.next().get( "n.prop" ), equalTo( Integer.toString( i ) ) );
+        }
+    }
```

Comment:
I think that we should also check the state of the database after the write query and that we have committed 100 times.